examples/llm/inference/hf_generate.py

```diff
@@ -0,0 +1,133 @@
+# Copyright 2022 MosaicML Examples authors
+# SPDX-License-Identifier: Apache-2.0
+import time
+import warnings
+from argparse import ArgumentParser, Namespace
+from contextlib import nullcontext
+
+import torch
+from composer.core import get_precision_context
+from composer.utils import get_device, reproducibility
+from transformers import AutoConfig, AutoModelForCausalLM, AutoTokenizer
+
+from examples.llm import MosaicGPT, MosaicGPTConfig
+
+
+def parse_args() -> Namespace:
+    """Parse commandline arguments."""
+    parser = ArgumentParser(
+        description='Load a HF CausalLM Model and use it to generate text.')
+    parser.add_argument('-n', '--name_or_path', type=str, required=True)
+    parser.add_argument(
+        '-p',
+        '--prompts',
+        nargs='+',
+        default=[
+            'My name is',
+            'This is an explanation of deep learning to a five year old. Deep learning is',
+        ])
+    parser.add_argument('--max_new_tokens', type=int, default=100)
+    parser.add_argument('--temperature', type=float, default=1.0)
+    parser.add_argument('--top_k', type=int, default=50)
+    parser.add_argument('--top_p', type=float, default=1.0)
+    parser.add_argument('--dtype',
+                        type=str,
+                        choices=['fp32', 'fp16', 'bf16'],
+                        default='bf16')
+    parser.add_argument('--autocast', action='store_true')
+    parser.add_argument('--device', type=str, default=None)
+    parser.add_argument('--seed', type=int, default=42)
+    return parser.parse_args()
+
+
+def main(args: Namespace) -> None:
+    reproducibility.seed_all(args.seed)
+
+    AutoConfig.register('mosaic_gpt', MosaicGPTConfig)
+    AutoModelForCausalLM.register(MosaicGPTConfig, MosaicGPT)
+
+    print('Loading HF model...')
+    model = AutoModelForCausalLM.from_pretrained(args.name_or_path)
+    model.eval()
+    print(f'n_params={sum(p.numel() for p in model.parameters())}')
+
+    print('\nLoading HF tokenizer...')
+    tokenizer = AutoTokenizer.from_pretrained(args.name_or_path)
+    if tokenizer.pad_token_id is None:
+        warnings.warn(
+            'pad_token_id is not set for the tokenizer. Using eos_token_id as pad_token_id.'
+        )
+        tokenizer.pad_token_id = tokenizer.eos_token_id
+    tokenizer.padding_side = 'left'
+
+    generate_kwargs = {
+        'max_new_tokens': args.max_new_tokens,
+        'temperature': args.temperature,
+        'top_p': args.top_p,
+        'top_k': args.top_k,
+        'use_cache': True,
```

Comment:
suggest making both of these args, and also adding an `eos_token_id` arg which defaults to `None`, and uses the tokenizer's eos token id